app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,24 @@
+require'pry'
+class SessionsController < ApplicationController
+  def new
+    redirect_to home_path if logged_in?
+  end
+
+  def create
+    user = User.find_by(email: params[:email])
+    if user && user.authenticate(params[:password])
+      session[:user_id] = user.id # rails conventions saves session information to cookies. Since cookies have limited size, we only save the user id. 
```

Comment:
I would not call it Rails convention. Session data is a cookie. In Sinatra app too.